From 3a4c9f9ae7b727004dcd04d07be01e944e18ca27 Mon Sep 17 00:00:00 2001
From: Tomasz Michalec <tomasz@marvell.com>
Date: Tue, 3 Mar 2020 15:10:37 +0100
Subject: [PATCH 500/767] octeontx2-serdes: Add CONFIG_OCTEONTX_SERDES

commit 85ee705c064453694a10f51e9fc98a2336dec86e from
git@git.assembla.com:cavium/WindRiver.linux.git

CONFIG_OCTEONTX_SERDES flag allows to exclude octeontx serdes diagnostic
commands from build.

Change-Id: Ifb08390df8298ce69e8251987a782cf091cf05ba
Signed-off-by: Tomasz Michalec <tomasz@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/24382
Reviewed-by: Chandrakala Chavva <cchavva@marvell.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/soc/marvell/Kconfig  | 7 +++++++
 drivers/soc/marvell/Makefile | 2 +-
 2 files changed, 8 insertions(+), 1 deletion(-)

diff --git a/drivers/soc/marvell/Kconfig b/drivers/soc/marvell/Kconfig
index 5cc6ac080ce6..3c81c2af9a93 100644
--- a/drivers/soc/marvell/Kconfig
+++ b/drivers/soc/marvell/Kconfig
@@ -116,4 +116,11 @@ config OCTEONTX2_EINJ
 	  called otx2-einj.
 	  Unless testing, say N here.
 
+config OCTEONTX_SERDES
+	tristate "OcteonTX serdes debug commands"
+	depends on ARM64
+	help
+	  Provides debugfs commands which allows to collect serdes information,
+	  capture eye or use PRBS.
+
 endmenu
diff --git a/drivers/soc/marvell/Makefile b/drivers/soc/marvell/Makefile
index d42d060a1269..f951e003b28b 100644
--- a/drivers/soc/marvell/Makefile
+++ b/drivers/soc/marvell/Makefile
@@ -1,6 +1,6 @@
 # SPDX-License-Identifier: GPL-2.0
 obj-y		+= octeontx2-rm/
 obj-y		+= octeontx2-dpi/
-obj-y		+= octeontx2-serdes/
+obj-$(CONFIG_OCTEONTX_SERDES) += octeontx2-serdes/
 obj-$(CONFIG_MDIO_DEBUGFS) += mdio_debugfs.o
 obj-y		+= octeontx2-ghes/
-- 
2.31.1

